src/golang/lib/job/databricks.go

```diff
@@ -0,0 +1,173 @@
+		}
+		storageConfig.S3Config.AWSAccessKeyID = j.conf.AwsAccessKeyID
+		storageConfig.S3Config.AWSSecretAccessKey = j.conf.AwsSecretAccessKey
+
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricks_lib.DatabricksFunctionScript, specStr, nil
+
+	} else if spec.Type() == ParamJobType {
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricks_lib.DatabricksParamScript, specStr, nil
+
+	} else if spec.Type() == AuthenticateJobType ||
+		spec.Type() == LoadJobType ||
+		spec.Type() == ExtractJobType ||
+		spec.Type() == LoadTableJobType ||
+		spec.Type() == DeleteSavedObjectsJobType ||
+		spec.Type() == DiscoverJobType {
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricks_lib.DatabricksDataScript, specStr, nil
+
+	} else if spec.Type() == SystemMetricJobType {
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricks_lib.DatabricksMetricScript, specStr, nil
+
+	} else {
+		return "", "", errors.New("Unsupported JobType was passed in.")
+	}
+}
+
+func (j *DatabricksJobManager) Config() Config {
+	return j.conf
+}
+
+func (j *DatabricksJobManager) Launch(
+	ctx context.Context,
+	name string,
+	spec Spec,
+) JobError {
+	log.Infof("Running %s job %s.", spec.Type(), name)
+
+	scriptFile, specStr, err := j.mapJobTypeToFile(spec)
+	if err != nil {
+		return systemError(err)
+	}
+	storageConfig, err := spec.GetStorageConfig()
+	if err != nil {
+		return systemError(errors.Wrap(err, "Spec unexpectedly has no storage config."))
+	}
+	bucket := storageConfig.S3Config.Bucket
+	pythonFilePath := fmt.Sprintf("%s/%s", bucket, scriptFile)
+
+	jobID, err := databricks_lib.CreateJob(ctx, j.databricksClient, name, j.conf.S3InstanceProfileARN, pythonFilePath)
+	if err != nil {
+		return systemError(errors.Wrap(err, "Error creating job in Databricks."))
+	}
+	runID, err := databricks_lib.RunNow(ctx, j.databricksClient, jobID, specStr)
+	if err != nil {
+		return systemError(errors.Wrap(err, "Error runnning job in Databricks."))
+	}
+	j.runMap[name] = runID
+	return nil
+}
+
+func (j *DatabricksJobManager) Poll(ctx context.Context, name string) (shared.ExecutionStatus, JobError) {
+	runId, ok := j.runMap[name]
+	if !ok {
+		return shared.UnknownExecutionStatus, jobMissingError(errors.New("Job doesn't exist."))
+	}
+
+	getRunReq := &jobs.GetRun{
+		RunId: runId,
+	}
+	getRunResp, err := j.databricksClient.Jobs.GetRun(ctx, *getRunReq)
+	if err != nil {
+		return shared.UnknownExecutionStatus, systemError(errors.Wrap(err, "Unable to get run from databricks."))
+	}
+
+	switch getRunResp.State.LifeCycleState {
```

Comment:
Would you want to decouple polling from RunNowAndWait on the Databricks SDK level?

https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.1.1/service/jobs#JobsAPI.RunNowAndWait